app/src/ingestion/scrapy_dst/spiders/ca_public_charge_spider.py

```diff
@@ -0,0 +1,104 @@
+import re
+
+from markdownify import markdownify
+from scrapy.http import HtmlResponse
+from scrapy.linkextractors import LinkExtractor
+from scrapy.selector import SelectorList
+from scrapy.spiders.crawl import CrawlSpider, Rule
+
+from src.util import string_utils  # noqa: E402
+
+AccordionSections = dict[str, list[str]]
+
+
+class CaPublicChargeSpider(CrawlSpider):
+    # This name is used on the commandline: scrapy crawl ca_public_charge_spider
+    name = "ca_public_charge_spider"
+    allowed_domains = ["keepyourbenefits.org"]
+    start_urls = ["https://keepyourbenefits.org/en/ca/public-charge"]
+
+    # This is used to substitute the base URL in the cache storage
+    common_url_prefix = "https://keepyourbenefits.org/en/ca/"
+
+    rules = (
+        Rule(
+            LinkExtractor(
+                allow=r"en/ca",
+                deny=(
+                    "es/ca/public-charge",  # has non-English content
+                    "media/",
+                    "_ima/",
+                    "use-the-guide",
+                    "find-help",
+                ),
+                allow_domains=allowed_domains,
+                deny_domains=("https://s3.amazonaws.com",),
+                canonicalize=True,
+                unique=True,
+            ),
+            callback="parse_page",
+            follow=True,
+        ),
+    )
+
+    def parse_page(self, response: HtmlResponse) -> dict[str, str | AccordionSections]:
+        extractions = {"url": response.url}
+        if len(response.css("h4::text").getall()) == 1:
+            title = response.css("h4.title::text").get()
+            extractions["title"] = title.strip()
+        else:
+            titles = ";".join(response.css("h4.title::text").getall())
+            extractions["title"] = titles
+        base_url = response.url
+
+        extractions |= self.parse_main_primary(base_url, response.css("div.module-full"))
+        extractions |= self.parse_main_content(base_url, response.css("div.GTM-1"))
+
+        return extractions
+
+    def to_markdown(self, base_url: str, html: str) -> str:
+        # convert larger text to header
+        larger_font_pattern = (
+            r'<p class="title fontsize-30 weightier colored text-center"([^>]*?)>(.*?)<\/p>'
+        )
+        html = re.sub(larger_font_pattern, r"<h3\1>\2</h3>", html)
+
+        # remove icon element
+        html = re.sub(r'<div class="ic-icon">(.*)</div>', "", html)
```

Comment:
Prefer to drop elements rather than use regex.
https://github.com/navapbc/labs-decision-support-tool/pull/178/files#diff-4af5d1278a2fc7ba0f6081d12642eb446d3db067c054b823bfbf50bf0a0d6aa2R64-R65

Why? https://stackoverflow.com/questions/590747/using-regular-expressions-to-parse-html-why-not